packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -152,14 +188,19 @@ export default class BranchPublish extends Command {
     }
     const projectId = await getProjectIdAsync(exp);
 
-    if (!name) {
+    if (!branchName && autoFlag) {
+      branchName =
+        (await vcs.getBranchNameAsync()) || `branch-${Math.random().toString(36).substr(2, 4)}`;
```

Comment:
We should create another task to make the `getBranchNameAsync` work inside GitHub Actions. The command this runs is `git rev-parse --abbrev-ref HEAD`, which works locally, but results in this error during a GitHub Action:

<img width="673" alt="Screen Shot 2021-08-06 at 12 06 27 AM" src="https://user-images.githubusercontent.com/6455018/128454575-056613bc-5382-4443-b677-85cffbfa5c82.png">

This is not a blocker for this PR, and this should generally not be a problem for CI, since we can pass in the branch name/message programmatically. But it'd be great to run `eas branch:publish --auto` locally, and then write the same command for the CI action.

Reply:
.git repository should exist after checkout, if it doesn't then I think it's bug in github action.

Seems like it might be cause by old git version in the image 
https://github.com/actions/checkout/issues/335
https://github.com/actions/checkout/issues/126

Reply:
@jonsamp let me know if @wkozyra95 suggestions for the github action don't work out.